lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -43,8 +43,30 @@ evaluateAndCacheCall(SILFunction &fn, SubstitutionMap substitutionMap,
 // ConstantFolding.h/cpp files should be subsumed by this, as this is a more
 // general framework.
 
+
+// We have a list of functions that we hackily special case.  In order to keep
+// this localized, we use this classifier function.  This should be replaced
+// with an attribute put on the standard library that captures this info.
+enum class WellKnownFunction {
+  Unknown,
+  StringInitEmpty,  // String.init()
+  AssertionFailure, // _assertionFailure(_:_:file:line:flags:)
+};
+
+
+static WellKnownFunction classifyFunction(StringRef mangledName) {
+  if (mangledName == "$SS2SycfC")
```

Comment:
Question: Is the long term solution to use `@_semantics(...)`/`@compilerEvaluable(...)` instead of hard-coding mangled names?

Reply:
Yes, exactly.  I mention this in the comment directly above this.